packages/vite-plugin-shopify-defer-css/src/index.ts

```diff
@@ -0,0 +1,186 @@
+import fs from 'node:fs'
+import path from 'node:path'
+import { AddressInfo } from 'node:net'
+import { Manifest, Plugin, ResolvedConfig, normalizePath } from 'vite'
+import { CSS_EXTENSIONS_REGEX, KNOWN_CSS_EXTENSIONS } from '../../vite-plugin-shopify/src/constants'
+import { assetUrl, viteEntryTag, viteTagDisclaimer, viteTagEntryPath, resolveDevServerUrl } from '../../vite-plugin-shopify/src/html'
+import { DevServerUrl } from '../../vite-plugin-shopify/src/types'
+
+/** Plugin configuration */
+export interface Options {
+  /**
+   * Root path to your Shopify theme directory.
+   *
+   * @default './'
+   */
+  themeRoot?: string
+
+  /**
+   * Front-end entry points directory.
+   *
+   * @default 'frontend/entrypoints'
+   */
+  entrypointsDir?: string
+
+  /**
+   * Front-end source code directory.
+   *
+   * @default 'frontend'
+   */
+  sourceCodeDir?: string
+
+  /**
+   * Specifies the file name of the snippet that loads your assets.
+   *
+   * @default 'vite-tag.liquid'
+   */
+  snippetFile?: string
+}
+
+// Generate a production stylesheet link tag for a style asset
+const stylesheetTag = (fileName: string, versionNumbers: boolean): string =>
+  `<link
+    rel="preload"
+    fetchpriority="{{ fetchpriority | default: 'low' }}"
+    href="{{ ${assetUrl(fileName, versionNumbers)} }}"
+    as="style"
+    onload="this.onload=null;this.rel='stylesheet'"
+   >
+   <noscript>
+     <link rel="stylesheet" href="{{ ${assetUrl(fileName, versionNumbers)} }}">
+   </noscript>`
+
+// Generate vite-tag snippet for development
+export const viteTagSnippetDev = (assetHost: string, entrypointsDir: string): string =>
+  `{% liquid
+  assign path_prefix = path | slice: 0
+  if path_prefix == '/'
+    assign file_url_prefix = '${assetHost}'
+  else
+    assign file_url_prefix = '${assetHost}/${entrypointsDir}/'
+  endif
+  assign file_url = path | prepend: file_url_prefix
+  assign file_name = path | split: '/' | last
+  if file_name contains '.'
+    assign file_extension = file_name | split: '.' | last
+  endif
+  assign css_extensions = '${KNOWN_CSS_EXTENSIONS.join('|')}' | split: '|'
+  assign is_css = false
+  if css_extensions contains file_extension
+    assign is_css = true
+  endif
+%}
+<script src="${assetHost}/@vite/client" type="module"></script>
+{% if is_css == true %}
+  <link
```

Comment:
@tuanpham-barrel it allows to defer css during dev mode

Reply:
It would keep the behavior consistent during dev and prod

Reply:
Would you want a different behavior during dev?

Reply:
The behavior between dev and prod is already different and dev cannot be used for testing performance so I don't think we have any benefit here.

Reply:
We can adjust here

Reply:
@tuanpham-barrel could you please adjust this with the desired behavior?